pkg/controllers/iamauthpolicy_controller.go

```diff
@@ -126,102 +131,29 @@ func (c *IAMAuthPolicyController) reconcileDelete(ctx context.Context, k8sPolicy
 }
 
 func (c *IAMAuthPolicyController) reconcileUpsert(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) (ctrl.Result, error) {
-	validationErr := c.validateSpec(ctx, k8sPolicy)
-	err := c.updateStatus(ctx, k8sPolicy, validationErr)
+	reason, err := c.ph.ValidateAndUpdateCondition(ctx, k8sPolicy)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	var statusPolicy model.IAMAuthPolicyStatus
-	if validationErr == nil {
-		modelPolicy := model.NewIAMAuthPolicy(k8sPolicy)
-		c.addFinalizer(k8sPolicy)
-		err = c.client.Update(ctx, k8sPolicy)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		statusPolicy, err = c.policyMgr.Put(ctx, modelPolicy)
-		if err != nil {
-			return reconcile.Result{}, services.IgnoreNotFound(err)
-		}
-		c.updateLatticeAnnotaion(k8sPolicy, statusPolicy.ResourceId, modelPolicy.Type)
+	if reason != policy.ReasonAccepted {
+		return ctrl.Result{}, nil
 	}
-	err = c.handleLatticeResourceChange(ctx, k8sPolicy, statusPolicy)
+	modelPolicy := model.NewIAMAuthPolicy(k8sPolicy)
+	c.addFinalizer(k8sPolicy)
+	err = c.client.Update(ctx, k8sPolicy)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	return ctrl.Result{}, nil
-}
-
-func (c *IAMAuthPolicyController) validateSpec(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) error {
-	tr := k8sPolicy.Spec.TargetRef
-	if tr.Group != gwv1beta1.GroupName {
-		return fmt.Errorf("%w: %s", GroupNameError, tr.Group)
-	}
-	if !slices.Contains([]string{"Gateway", "HTTPRoute", "GRPCRoute"}, string(tr.Kind)) {
-		return fmt.Errorf("%w: %s", KindError, tr.Kind)
-	}
-	refExists, err := c.targetRefExists(ctx, k8sPolicy)
+	statusPolicy, err := c.pm.Put(ctx, modelPolicy)
 	if err != nil {
-		return err
-	}
-	if !refExists {
-		return fmt.Errorf("%w: %s", TargetRefNotFound, tr.Name)
+		return reconcile.Result{}, services.IgnoreNotFound(err)
 	}
-	conflictingPolicies, err := c.findConflictingPolicies(ctx, k8sPolicy)
+	c.updateLatticeAnnotaion(k8sPolicy, statusPolicy.ResourceId, modelPolicy.Type)
+	err = c.handleLatticeResourceChange(ctx, k8sPolicy, statusPolicy)
```

Comment:
actually it follows our naming convention - https://github.com/aws/aws-application-networking-k8s/blob/main/docs/contributing/developer-cheat-sheet.md#service-service-or-service

I extended that to status too, because it's same as modelPolicy